addons/account/views/ir_actions_views.xml

```diff
@@ -6,7 +6,7 @@
         <field name="inherit_id" ref="base.act_report_xml_view"/>
         <field name="arch" type="xml">
             <field name="paperformat_id" position="after">
-                <field name="is_invoice_report"/>
+                <field name="is_invoice_report" invisible="'model' != 'account.move'"/>
```

Comment:
Doing just this will hide the problem if `is_invoice_report` is already incorrectly set.

In addition to this, I would suggest that you also modify the `_is_invoice_report` method to check that the report model is `account.move`.